cli/wallet.go

```diff
@@ -743,3 +745,69 @@ var walletMarketAdd = &cli.Command{
 		return nil
 	},
 }
+
+var WalletIDSp = &cli.Command{
+	Name:      "IDsp",
+	Usage:     "Generate a non-interactive authentication token based on worker address which can be used to establish SP operator identity",
+	ArgsUsage: "<miner address>",
+	Action: func(cctx *cli.Context) error {
+		api, closer, err := GetFullNodeAPIV1(cctx)
+		if err != nil {
+			return xerrors.Errorf("getting node API: %w", err)
+		}
+		defer closer()
+		ctx := ReqContext(cctx)
+
+		afmt := NewAppFmt(cctx.App)
+
+		if !cctx.Args().Present() {
+			return fmt.Errorf("must pass miner id")
+		}
+
+		maddr, err := address.NewFromString(cctx.Args().First())
```

Comment:
`maddr` usually means `MultiAddress` in the codebase, just `addr` ?

Reply:
This pattern I picked up from lotus code itself. They typically use maddr for miner address to distinguish it AFAIK